app/controllers/application_controller.rb

```diff
@@ -430,8 +430,12 @@ def all_identifiers
     ].filter { |e| e && !e.to_s.empty? }.uniq
   end
 
-  def after_sign_in_path_for(_user)
-    @after_login_path || hub_assigned_clients_path
+  def after_sign_in_path_for(user)
```

Comment:
This hook fires after the person has already been signed in, so they can actually navigate around the app and ignore the force password reset page

If we want to lock people off from doing stuff before their password has been reset, we either need to:
* add a `before_action` to all authenticated pages making sure people get redirected back to the password reset (would probably require keeping an extra bit on info on the user saying they still need to reset their password, because after the initial login we don't have it anymore)
OR,
* find a different way in the code to redirect, where we have access to the password and have confirmed it is correct but the client is not devise-authenticated (not sure if this is possible or not!)

Reply:
Resolved in 3f16554f4

Reply:
We took the second path of checking the password _after_ sign-in (and updated the `forced_password_reset_at` value to prevent checking it again).

In order to go with that suggestion of doing it _before_ Devise has authenticated the user, it _should_ require some sort of nonce or state value so we could avoid access to the password reset page from a non-privileged user.